quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs

```diff
@@ -535,6 +544,18 @@ impl DocMapper for DefaultDocMapper {
 
         if let Some(dynamic_field) = self.dynamic_field {
             if !dynamic_json_obj.is_empty() {
+                if !self.concate_dynamic_fields.is_empty() {
+                    let json_obj_values = dynamic_json_obj
+                        .iter()
+                        .flat_map(|(_key, val)| JsonValueIterator::new(val.clone()))
+                        .flat_map(map_primitive_json_to_tantivy);
+
+                    for value in json_obj_values {
+                        for concate_dynamic_field in &self.concate_dynamic_fields {
+                            document.add_field_value(*concate_dynamic_field, value.clone());
```

Comment:
Do you know a simple trick to remove the last clone? 

I haven't found anything simple in the library.
It feels sad if we have a single concate_dynamic_fields (the most common case)

Reply:
I couldn't find a good way to do that, so I created a custom iterator which does the trick (`ZipCloneable`). Let me know if you think that's better or not